crates/hyperqueue/src/client/commands/wait.rs

```diff
@@ -72,18 +73,23 @@ pub async fn wait_for_jobs(
 
 pub async fn wait_for_jobs_with_progress(
     session: &mut ClientSession,
-    mut jobs: Vec<JobInfo>,
+    jobs: &[JobInfo],
 ) -> anyhow::Result<()> {
-    jobs.retain(|info| !is_terminated(info));
-
-    if jobs.is_empty() {
+    if jobs.iter().all(is_terminated) {
         log::warn!("There are no jobs to wait for");
     } else {
-        let total_tasks: JobTaskCount = jobs.iter().map(|info| info.n_tasks).sum();
-        let mut remaining_job_ids: Set<JobId> = jobs.into_iter().map(|info| info.id).collect();
+        let total_tasks: JobTaskCount = jobs
+            .iter()
+            .filter(|info| !is_terminated(info))
+            .map(|info| info.n_tasks)
+            .sum();
+        let mut remaining_job_ids: BTreeSet<JobId> = jobs
```

Comment:
Any particular reason why `Set` was switched to `BTreeSet` and why the `Vec` was changed to a slice?

Reply:
Because we need to random removes & sorted iterator for creating IntArray.